gateway/modules/filestore/gcpstorage/create.go

```diff
@@ -3,14 +3,20 @@ package gcpstorage
 import (
 	"context"
 	"io"
+	"strings"
 
 	"github.com/spaceuptech/space-cloud/gateway/model"
 	"github.com/spaceuptech/space-cloud/gateway/utils"
 )
 
 // CreateFile creates a file in GCPStorage
 func (g *GCPStorage) CreateFile(req *model.CreateFileRequest, file io.Reader) error {
-	wc := g.client.Bucket(g.bucket).Object(utils.JoinLeading(req.Path, req.Name, "/")).NewWriter(context.TODO())
+	req.Path = strings.TrimPrefix(req.Path, "/")
+	path := req.Path + "/" + req.Name
+	if len(req.Path) == 0 {
+		path = req.Name
+	}
```

Comment:
Logic is incorrect

Reply:
Look carefully I am checking the length of req.path after trim prefix.
if user only provides / . then because of trim prefix req.path will be empty in that case we have to just create a file inside the root directory